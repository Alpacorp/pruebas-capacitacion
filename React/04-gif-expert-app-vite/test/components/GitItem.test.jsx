const { render, screen } = require("@testing-library/react");
const { GifItem } = require("../../src/components/GifItem");

describe("Pruebas en <GifItem />", () => {
  const title = "The last of us";
  const url = "https://i.gifer.com/1ZQ.gif";

  test("Debe de renderizar correctamente el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
    // screen.debug();
  });

  test("Debe de mostrar el tÃ­tulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
