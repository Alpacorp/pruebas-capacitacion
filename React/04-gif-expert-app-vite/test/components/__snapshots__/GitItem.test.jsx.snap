// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de renderizar correctamente el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="The last of us"
      src="https://i.gifer.com/1ZQ.gif"
    />
    <p>
      The last of us
    </p>
  </div>
</div>
`;
